library/CMakeLists.txt

```diff
@@ -192,7 +192,20 @@ if(CMAKE_COMPILER_IS_GNUCC)
 endif(CMAKE_COMPILER_IS_GNUCC)
 
 if(CMAKE_COMPILER_IS_CLANG)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-declarations -Wmissing-prototypes -Wdocumentation -Wno-documentation-deprecated-sync -Wunreachable-code")
+    execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE CLANG_VERSION)
+
+    # Clang version 15 or above introduces new rules for -WDocumentation that
+    # state that we cannot have a non-empty description after \return, \retval
+    # VALUE or param NAME. However, the Doxygen documentation says that the
+    # description runs to the end of the paragraph and doesn't say that it can't
+    # be empty, and the Doxygen tool is happy with our code. With
+    # -WDocumentation turned on using clang >= 15.0, you will see lots of errors
+    # along the lines of: "error: empty paragraph passed to '\retval'".
+    if(CLANG_VERSION VERSION_GREATER 15.0 OR CLANG_VERSION VERSION_EQUAL 15.0)
```

Comment:
In `development`, we can use `VERSION_GREATER_EQUAL`, but 2.28 supports older cmake that lack this operator. Alternatively, we could invert the test and write `VERSION_LESS 15.0`. Or we can keep what's here.

Reply:
Yup, since it's an if-else much easier to invert the test and swap the lines around